### PR TITLE
Add code from Week 1

### PR DESCRIPTION
**Pull request by Koch-Dustin**
Changes:
1. Add Mario_less
2. Add Mario_more
3. Add Cash
4. Add Credit
5. Add Hello

### CODE REVIEW

File: Woche1/credit/credit.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        }else if((firstTwoDigits == 34 || firstTwoDigits == 37) && count == 15) {
+            printf("AmericanExpress\n");
+        }else if((count == 13 || count == 16) && firstDigit == 4) {
+            printf("Visa\n");
+        }else {
+            printf("Invalid\n");
+        }
+
+    }else {
+        printf("Invalid\n");
```
Comment (shoreal) [inline Woche1/credit/credit.c:56]:
```suggestion
        } else if((firstTwoDigits == 34 || firstTwoDigits == 37) && count == 15) {
            printf("AmericanExpress\n");
        } else if((count == 13 || count == 16) && firstDigit == 4) {
            printf("Visa\n");
        } else {
            printf("Invalid\n");
        }

    } else {
        printf("Invalid\n");
```

File: Woche1/credit/credit.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+}
```
Comment (shoreal) [inline Woche1/credit/credit.c:59]: EOF missing (deutsch: End of File fehlt, also Leerzeile am Ende der Datei fehlt). Bitte noch hinzufügen.

File: Woche1/finished_cash/cash.c [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+int get_cents(void);
+int calculate_quarters(int cents);
+int calculate_dimes(int cents);
+int calculate_nickels(int cents);
+int calculate_pennies(int cents);
+
+int main(void)
+{
+    // Ask how many cents the customer is owed
```
Comment (shoreal) [inline Woche1/finished_cash/cash.c:12]:
```suggestion
```
Remove comments, we don't use them ever :D 

We are trying the approach of writing code, which is easily understandable without comments. The motto is: if you need comments to explain your code, your code isn't clean enough.

We only use comments in rare cases.

File: Woche1/finished_cash/cash.c [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    // Calculate the number of pennies to give the customer
+    int pennies = calculate_pennies(cents);
+    cents = cents - pennies * 1;
+
+    // Sum coins
+    int coins = quarters + dimes + nickels + pennies;
+
+    // Print total number of coins to give the customer
+    printf("%i\n", coins);
```
Comment (shoreal) [inline Woche1/finished_cash/cash.c:35]:
```suggestion
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    int coins = quarters + dimes + nickels + pennies;

    printf("%i\n", coins);
```

File: Woche1/finished_cash/cash.c [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+{
+    int nickels = 5;
+    return cents / nickels;
+}
+
+int calculate_pennies(int cents)
+{
+    int pennies = 1;
+    return cents / pennies;
+}
```
Comment (shoreal) [inline Woche1/finished_cash/cash.c:69]: EOF missing

File: Woche1/finished_credit/credit.c [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+            printf("Visa\n");
+        }else {
+            printf("Invalid\n");
+        }
+
+    }else {
+        printf("Invalid");
+    }
+
+}
```
Comment (shoreal) [inline Woche1/finished_credit/credit.c:59]: EOF missing

File: Woche1/finished_hello/hello.c [hunk 7]
```diff
@@ -0,0 +1,8 @@
+#include <cs50.h>
+#include <stdio.h>
+
+int main(void)
+{
+    string name = get_string("Wie heißt du?: ");
+    printf("hello, %s\n", name);
+}
```
Comment (shoreal) [inline Woche1/finished_hello/hello.c:8]: EOF missing

File: Woche1/finished_mario_less/mario-less.c [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+        for(int j = 0; j < i + 1; j++) {
+            printf("#");
+        }
+        printf("\n");
+
+    }
+
+
+}
```
Comment (shoreal) [inline Woche1/finished_mario_less/mario-less.c:25]: EOF missing

File: Woche1/finished_mario_more/mario-more.c [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+        for(int j = -1; j < i; j++) {
+            printf("#");
+        }
+
+        printf("\n");
+
+    }
+
+
+}
```
Comment (shoreal) [inline Woche1/finished_mario_more/mario-more.c:31]: EOF missing

File: Woche1/finished_mario_more/mario-more.c [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+        printf("  ");
+
+        for(int j = -1; j < i; j++) {
+            printf("#");
+        }
+
+        printf("\n");
+
+    }
+
```
Comment (shoreal) [inline Woche1/finished_mario_more/mario-more.c:29]:
```suggestion
    }
```

**Review by shoreal — CHANGES_REQUESTED** [review]

File: Woche1/finished_mario_less/mario-less.c [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+            printf(" ");
+        }
+
+        for(int j = 0; j < i + 1; j++) {
+            printf("#");
+        }
+        printf("\n");
+
+    }
+
```
Comment (shoreal) [inline Woche1/finished_mario_less/mario-less.c:23]:
```suggestion
```

**Review by shoreal — CHANGES_REQUESTED** [review]

**Comment by shoreal** [comment]
I would suggest to configure your IDE (VSCode) to automatically add eof's (blank lines at the end of the file).

File: Woche1/finished_mario_less/mario-less.c [hunk 12]
```diff
@@ -0,0 +1,24 @@
+#include <cs50.h>
+#include <stdio.h>
+
+int main(void)
+{
+    int n;
+    do {
```
Comment (stefanluth) [inline Woche1/finished_mario_less/mario-less.c:7]:
```suggestion
    int n;
    
    do {
```

File: Woche1/finished_mario_less/mario-less.c [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+            printf(" ");
+        }
+
+        for(int j = 0; j < i + 1; j++) {
+            printf("#");
+        }
+        printf("\n");
+
+    }
+
```
Comment (stefanluth) [inline Woche1/finished_mario_less/mario-less.c:23]:
```suggestion
    }
```

File: Woche1/finished_mario_more/mario-more.c [hunk 14]
```diff
@@ -0,0 +1,30 @@
+#include <cs50.h>
+#include <stdio.h>
+
+int main(void)
+{
+    int n;
+    do {
```
Comment (stefanluth) [inline Woche1/finished_mario_more/mario-more.c:7]:
```suggestion
    int n;
    
    do {
```

File: Woche1/finished_mario_more/mario-more.c [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+        printf("  ");
+
+        for(int j = -1; j < i; j++) {
+            printf("#");
+        }
+
+        printf("\n");
+    }
+
+
```
Comment (stefanluth) [inline Woche1/finished_mario_more/mario-more.c:29]:
```suggestion
```

**Review by stefanluth — CHANGES_REQUESTED** [review]

File: Woche1/credit/credit.c [hunk 16]
```diff
@@ -0,0 +1,59 @@
+#include <cs50.h>
+#include <stdio.h>
+
+int main(void) {
+
```
Comment (DevBastiS) [inline Woche1/credit/credit.c:5]:
```suggestion
```

File: Woche1/credit/credit.c [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+    long long ccard;
+    long divisor = 10;
+
+    do {
+        ccard = get_long_long("Credit: ");
+    } while (ccard <= 0);
+    
+    long long workingCC = ccard;
```
Comment (DevBastiS) [inline Woche1/credit/credit.c:13]: Isn't one long enough?

Comment (DevBastiS) [reply Woche1/credit/credit.c:13]: Also what does "workingCC" mean?

File: Woche1/credit/credit.c [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+        workingCC = workingCC / 100;
+    }
+
+    workingCC = ccard;
+    while (workingCC != 0) {
+        workingCC = workingCC / 10;
+        count++;
+    }
+    
+    for(int i = 0; i < count - 2; i++) {
```
Comment (DevBastiS) [inline Woche1/credit/credit.c:37]:
```suggestion
    for (int i = 0; i < count - 2; i++) {
```

File: Woche1/credit/credit.c [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+        divisor = divisor * 10;
+    }
+
+    int firstDigit = ccard / divisor;
+    int firstTwoDigits = ccard / (divisor / 10); 
+
+    if(sum % 10 == 0) {
```
Comment (DevBastiS) [inline Woche1/credit/credit.c:44]:
```suggestion
    if (sum % 10 == 0) {
```

File: Woche1/credit/credit.c [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+        if(count == 16 && (firstTwoDigits == 51 || firstTwoDigits == 52 || firstTwoDigits == 53 || firstTwoDigits == 54 || firstTwoDigits == 55)) {
```
Comment (DevBastiS) [inline Woche1/credit/credit.c:45]: This if conditions are very hard to read. Try it this way:

```
bool CardIsMasterCard = <condition>;
if (CardIsMasterCard) {
  ....
}
```
This way the reader can read over the if clause like it is an english sentence.

Comment (DevBastiS) [reply Woche1/credit/credit.c:45]:
```suggestion
        if (count == 16 && (firstTwoDigits == 51 || firstTwoDigits == 52 || firstTwoDigits == 53 || firstTwoDigits == 54 || firstTwoDigits == 55)) {
```

File: Woche1/credit/credit.c [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+            printf("MasterCard\n");
+        } else if((firstTwoDigits == 34 || firstTwoDigits == 37) && count == 15) {
```
Comment (DevBastiS) [inline Woche1/credit/credit.c:47]:
```suggestion
        } else if ((firstTwoDigits == 34 || firstTwoDigits == 37) && count == 15) {
```

File: Woche1/credit/credit.c [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+            printf("AmericanExpress\n");
+        } else if((count == 13 || count == 16) && firstDigit == 4) {
```
Comment (DevBastiS) [inline Woche1/credit/credit.c:49]:
```suggestion
        } else if ((count == 13 || count == 16) && firstDigit == 4) {
```

File: Woche1/credit/credit.c [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+            printf("Visa\n");
+        } else {
+            printf("Invalid\n");
+        }
+
+    } else {
```
Comment (DevBastiS) [inline Woche1/credit/credit.c:55]: That if statement seem kinda long^^ Maybe you can use "early exit" here?

File: Woche1/credit/credit.c [hunk 24]
```diff
@@ -0,0 +1,59 @@
...
+        printf("Invalid\n");
+    }
+
```
Comment (DevBastiS) [inline Woche1/credit/credit.c:58]:
```suggestion
```

File: Woche1/finished_cash/cash.c [hunk 25]
```diff
@@ -0,0 +1,62 @@
...
+int calculate_nickels(int cents) {
+    int nickels = 5;
+
+    return cents / nickels;
+}
+
+int calculate_pennies(int cents) {
+    int pennies = 1;
+
+    return cents / pennies;
```
Comment (DevBastiS) [inline Woche1/finished_cash/cash.c:61]: you could just return "cents" here.

File: Woche1/finished_mario_less/mario-less.c [hunk 26]
```diff
@@ -0,0 +1,24 @@
...
+    for (int i = 0; i < n; i++) {
+
+        for(int j = 0; j < n - i; j++) {
+            printf(" ");
+        }
+
+        for(int j = 0; j < i + 1; j++) {
+            printf("#");
+        }
```
Comment (DevBastiS) [inline Woche1/finished_mario_less/mario-less.c:20]:
```suggestion
        }
        
```

File: Woche1/finished_mario_less/mario-less.c [hunk 27]
```diff
@@ -0,0 +1,24 @@
...
+        printf("\n");
+    }
+
```
Comment (DevBastiS) [inline Woche1/finished_mario_less/mario-less.c:23]:
```suggestion
```

File: Woche1/finished_mario_more/mario-more.c [hunk 28]
```diff
@@ -0,0 +1,30 @@
...
+        }
+        printf("  ");
+
+        for(int j = -1; j < i; j++) {
+            printf("#");
+        }
+
+        printf("\n");
+    }
+
```
Comment (DevBastiS) [inline Woche1/finished_mario_more/mario-more.c:29]:
```suggestion
```

**Review by DevBastiS — CHANGES_REQUESTED** [review]

File: Woche1/credit/credit.c (hunk 17 above)
Comment (Koch-Dustin) [reply Woche1/credit/credit.c:13]: "workingCC" is the input of the user that matches the condition.

**Review by stefanluth — APPROVED** [review]

**Review by shoreal — APPROVED** [review]

**Pull request merged**